rio_stac/stac.py

```diff
@@ -0,0 +1,197 @@
+    metadata: Dict[str, Any] = {}
+
+    metadata["name"] = src_dst.name
+
+    # To Do: handle non-geo data
+    bbox = warp.transform_bounds(
+        src_dst.crs, "epsg:4326", *src_dst.bounds, densify_pts=21
+    )
+
+    metadata["bbox"] = list(bbox)
+    metadata["footprint"] = bbox_to_geom(bbox)
+
+    # GDAL Metadata
+    metadata["tags"] = src_dst.tags()
+    metadata["width"] = src_dst.width
+    metadata["height"] = src_dst.height
+    metadata["dtype"] = src_dst.dtypes[0]
+
+    # Proj
+    try:
+        epsg = src_dst.crs.to_epsg() or None
+    except AttributeError:
+        epsg = None
+
+    metadata["proj"] = {
+        "epsg": epsg,
+        "geometry": bbox_to_geom(src_dst.bounds),
+        "bbox": list(src_dst.bounds),
+        "shape": [src_dst.height, src_dst.width],
+        "transform": list(src_dst.transform),
+    }
+
+    # rasterio can't reproduce wkt2
+    # ref: https://github.com/mapbox/rasterio/issues/2044
+    # if epsg is None:
+    #      metadata["proj"]["wkt2"] = src_dst.crs.to_wkt()
+
+    return metadata
+
+
+def get_media_type(
+    src_dst: Union[DatasetReader, DatasetWriter, WarpedVRT, MemoryFile],
+    input_media_type: Optional[pystac.MediaType] = None,
+) -> Optional[pystac.MediaType]:
+    """Define or validate MediaType."""
+    driver = src_dst.driver
+
+    output_media_type = input_media_type
+    if driver == "GTiff":
+        if src_dst.crs:
+            output_media_type = pystac.MediaType.GEOTIFF
+        else:
+            output_media_type = pystac.MediaType.TIFF
+
+    elif driver in [
+        "JP2ECW",
+        "JP2KAK",
+        "JP2LURA",
+        "JP2MrSID",
+        "JP2OpenJPEG",
+        "JPEG2000",
+    ]:
+        output_media_type = pystac.MediaType.JPEG2000
+
+    elif driver in ["HDF4", "HDF4Image"]:
+        output_media_type = pystac.MediaType.HDF
+
+    elif driver in ["HDF5", "HDF5Image"]:
+        output_media_type = pystac.MediaType.HDF5
+
+    elif driver == "JPEG":
+        output_media_type = pystac.MediaType.JPEG
+
+    elif driver == "PNG":
+        output_media_type = pystac.MediaType.PNG
+
+    if not output_media_type:
+        warnings.warn("Could not determine the media type from GDAL driver.")
+        return input_media_type
+
+    if input_media_type == pystac.MediaType.COG and driver != "GTiff":
+        raise Exception("Input file is not a GeoTIFF.")
+
+    elif input_media_type and output_media_type != input_media_type:
+        raise Exception("MediaType do not match.")
+
+    return output_media_type
+
+
+def create_stac_item(
+    source: Union[str, DatasetReader, DatasetWriter, WarpedVRT, MemoryFile],
+    input_datetime: Optional[datetime.datetime] = None,
+    extensions: Optional[List[str]] = None,
+    collection: Optional[str] = None,
+    item_properties: Optional[Dict] = None,
+    id: Optional[str] = None,
+    asset_name: str = "asset",
+    asset_roles: Optional[List[str]] = None,
+    asset_media_type: Optional[pystac.MediaType] = None,
```

Comment:
media_type is not required from the spec POV. 

I've added `get_media_type` to **find** and **validate** input media type if given.

The gotcha is that if the GeoTIFF is a COG we can't tag it with pystac.MediaType.COG because we can't know it for sure without using rio-cogeo. 

I think this is OK but we should make sure to write it down in the docs.

Reply:
or maybe we get rid of it and just let the user pass the media_type 🤷 

cc @kylebarron

Reply:
see #4